src/loo_nonfactorizable.jl

```diff
@@ -0,0 +1,68 @@
+function diag_cov_from_cov_chol(F::Cholesky)
+    if F.uplo == 'U'
+        Uinv = inv(UpperTriangular(F.factors))
+        return map(Base.Fix1(sum, abs2), eachrow(Uinv))
+    else
+        Linv = inv(LowerTriangular(F.factors))
+        return map(Base.Fix1(sum, abs2), eachcol(Linv))
+    end
+end
+
+whitened_residuals(F::Cholesky, r::AbstractVector) = F \ r
+whitened_residuals(F::AbstractMatrix, r::AbstractVector) = F * r
+
+function pointwise_normal_loglikelihood!(
+    out::AbstractArray,
+    mean::AbstractVector,
+    obs::AbstractVector,
+    F::Union{Cholesky,AbstractMatrix},
+    c::AbstractVector,
+)
+    r = obs - mean
+    g = whitened_residuals(F, r)
+    @. out = (log(c) - g^2 / c - log(2π)) / 2
+    return nothing
+end
+
+function elements_from_d(d::Distributions.MvNormal)
+    μ = d.μ
+    F = d.Σ.chol
+    c = diag_cov_from_cov_chol(F)
+    return μ, F, c
+end
+
+function elements_from_d(d::Distributions.MvNormalCanon)
+    μ = d.μ
+    iΣ = d.J
+    c = diag(iΣ)
+    return μ, iΣ, c
+end
```

Comment:
These names aren't very informative, but more importantly, I think it's not necessary for these two to have this utility method at all, e.g. here's an implementation I had come up with:
```julia
function pointwise_loglikelihood!(logl::AbstractVector{<:Real}, dist::MvNormal, y::AbstractVector{<:Real})
    Σ = cov(dist)
    λ = _pd_diag_inv(Σ)
    g = Σ \ (y - mean(dist))
    return @. logl = (log(λ) - g^2 / λ - log2π) / 2
end
function pointwise_loglikelihood!(logl::AbstractVector{<:Real}, dist::MvNormalCanon, y::AbstractVector{<:Real})
    J = invcov(dist)
    λ = diag(J)
    cov_inv_y = J * y
    return @. logl = (log(λ) - (cov_inv_y - dist.h)^2 / λ - log2π) / 2
end
```